tests/unit/packaging/test_utils.py

```diff
@@ -32,6 +34,24 @@ def test_simple_detail_empty_string(db_request):
     assert expected_content["files"][0]["requires-python"] is None
 
 
+def test_simple_detail_with_provenance(
+    db_request, integrity_service, dummy_attestation
+):
+    project = ProjectFactory.create()
+    release = ReleaseFactory.create(project=project, version="1.0")
+    file = FileFactory.create(release=release)
+
+    db_request.route_url = lambda *a, **kw: "the-url"
+    db_request.oidc_publisher = GitHubPublisherFactory.create()
+
+    provenance = integrity_service.build_provenance(
+        db_request, file, [dummy_attestation]
+    )
```

Comment:
```suggestion

    # Provenance objects are generated at upload time, so we
    # are mocking it here.
    provenance = integrity_service.build_provenance(
        pretend.stub(
            oidc_publisher=GitHubPublisherFactory.create()
        ), file, [dummy_attestation]
    )
```

I think we should not reuse the `db_request` object in the `integrity_service` because this is supposed to be during upload time.

Reply:
True, good call!